src/HttpClientPoolItem.php

```diff
@@ -0,0 +1,127 @@
+<?php
+
+namespace Http\Client\Common;
+
+use Psr\Http\Message\RequestInterface;
+use Http\Client\Exception;
+
+/**
+ * A HttpClientPoolItem represent a HttpClient inside a Pool.
+ *
+ * It is disabled when a request failed and can be reenable after a certain number of seconds
+ * It also keep tracks of the current number of request the client is currently sending (only usable for async method)
+ *
+ * @author Joel Wurtz <joel.wurtz@gmail.com>
+ */
+class HttpClientPoolItem extends FlexibleHttpClient
```

Comment:
Why not? It can be maintained it in this client:

``` php
<?php

final class HttpClientPoolItem
{
    private $client;

    public function __construct($client, $reenableAfter = null)
    {
        $this->client = new FlexibleHttpClient($client);

        $this->reenableAfter = $reenableAfter;
    }
}
```